### PR TITLE
[E-idora-create < T464-MAGE] Implement set rel property for create functions

### PR DESCRIPTION
**Pull request by ind1xa**
### Description

Implemented create.set_rel_property and e2e correctness tests for it

### Pull request type

- [ ] Bugfix
- [x ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/create_module/algorithm/create.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          throw mgp::ValueException("First argument must be a relationsip, relationship's id or a list of those.");
+        }
+      }
+    } else if (arguments[0].IsRelationship()) {
+      relIds.insert(std::move(arguments[0].ValueRelationship().Id()));
+    } else if (arguments[0].IsInt()) {
+      relIds.insert(std::move(mgp::Id::FromInt(arguments[0].ValueInt())));
+    } else {
+      throw mgp::ValueException("First argument must be a relationsip, relationship's id or a list of those.");
+    }
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:27]: Same question for isolathing this part

**Review by antoniofilipovic — COMMENTED** [review]
One discission point

File: cpp/create_module/algorithm/create.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  } else if (argument.IsRelationship()) {
+    result_set.insert(std::move(argument.ValueRelationship().Id()));
+  } else if (argument.IsInt()) {
+    result_set.insert(std::move(mgp::Id::FromInt(argument.ValueInt())));
+  } else {
+    throw mgp::ValueException("First argument must be a relationsip, relationship's id or a list of those.");
+  }
+  return result_set;
+}
+
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:24]: Also this one: https://github.com/memgraph/mage/pull/273

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Refactor one function according to linked PR

File: cpp/create_module/algorithm/create.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  if (element.IsRelationship()) {
+    auto rel = element.ValueRelationship();
+    SetElementProp(rel, prop_key_list, prop_value_list, record_factory);
+    return;
+  }
+  if (element.IsInt()) {
+    relIds.insert(mgp::Id::FromInt(element.ValueInt()));
+    return;
+  }
+  throw mgp::ValueException("First argument must be node, relationship, id or a list of those.");
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:24]: Where are we processing nodes here?

File: cpp/create_module/algorithm/create.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    if (arguments[0].IsList()) {
+      for (const auto element : arguments[0].ValueList()) {
+        ProcessElement(element, graph, prop_key_list, prop_value_list, record_factory, relIds);
+      }
+    } else {
+      ProcessElement(arguments[0], graph, prop_key_list, prop_value_list, record_factory, relIds);
+    }
+    for (auto rel : graph.Relationships()) {
+      if (relIds.contains(rel.Id())) {
```
Comment (antoniofilipovic) [inline cpp/create_module/algorithm/create.cpp:48]: What if relIds is empty? I think we can make that check before and do early return?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
One structural change

File: cpp/create_module/algorithm/create.cpp (hunk 3 above)
Comment (ind1xa) [reply cpp/create_module/algorithm/create.cpp]: In last version of these procedures I wanted to use the same ProcessElement for all create functions when merging so I wrote universal exception message but now its not possible anymore but I forgot to change the message haha

**Review by antoniofilipovic — APPROVED** [review]
Looks good to me

**Pull request merged**